### PR TITLE
Add support for exp(::Quaternion{Int})

### LINKED ISSUE #39: in exp() function, eps yields an error when used on an integer
Using the quaternion package and instantiating quaternions of the shape
a = Quaternion(1,1,1,1)
the command
exp(a) 
returned an error of the function eps : "no method matching eps(::Int64)". Maybe an automatic convert of the coordinates of the quaternion to Float could solve the problem? 
In the meantime, I just changed my initialization for:
a = Quaternion(1.0,1.0,1.0,1.0) which lifted the error.

### PR DESCRIPTION
**Pull request by hyrodium**
This PR fixes #39.

### CODE REVIEW

File: src/Quaternion.jl [hunk 1]
```diff
@@ -118,7 +118,7 @@ function exp(q::Quaternion)
     if th > 0
         scale *= sin(th) / th
     end
-    Quaternion(se * cos(th), scale * q.v1, scale * q.v2, scale * q.v3, abs(s) < eps(typeof(s)))
+    Quaternion(se * cos(th), scale * q.v1, scale * q.v2, scale * q.v3, abs(s) < eps(typeof(se)))
```
[the remark this answers is not in the document]
Comment (sethaxen) [reply src/Quaternion.jl:121]: Yes, I think `iszero` is more reasonable. Nowhere else in the code base is approximate equality used to establish whether a quaternion is normalized, and `norm` indicates structural normalization, which I think fits better with exact equality.

[the remark this answers is not in the document]
Comment (sethaxen) [reply src/Quaternion.jl:121]: It's a little grey whether this would be a breaking change, but because this line of code is currently inconsistent with how `norm` is interpreted elsewhere in the codebase, I think this is a fine change to make.

[the remark this answers is not in the document]
Comment (hyrodium) [reply src/Quaternion.jl]: Thanks for the review! I've updated with `iszero`.

File: src/Quaternion.jl [hunk 2]
```diff
@@ -118,7 +118,7 @@ function exp(q::Quaternion)
     if th > 0
         scale *= sin(th) / th
     end
-    Quaternion(se * cos(th), scale * q.v1, scale * q.v2, scale * q.v3, abs(s) < eps(typeof(s)))
+    Quaternion(se * cos(th), scale * q.v1, scale * q.v2, scale * q.v3, iszero(abs(s)))
```
Comment (sethaxen) [inline src/Quaternion.jl:121]: No need for `abs` here:
```suggestion
    Quaternion(se * cos(th), scale * q.v1, scale * q.v2, scale * q.v3, iszero(s))
```

**Review by sethaxen — CHANGES_REQUESTED** [review]
One more thing, can you add a test for `exp` when `s` is zero and not zero, along with a check that the norm is 1, not 1, respectively?

File: src/Quaternion.jl (hunk 2 above)
Comment (hyrodium) [reply src/Quaternion.jl]: Oh, that's my mistake. Thanks for the suggestion!

**Review by sethaxen — APPROVED** [review]
